### PR TITLE
community[minor]: added a feature to filter documents in Mongoloader

### PR DESCRIPTION
**Pull request by iamKushagra**
"community: added a feature to filter documents in Mongoloader"
    - **Description:** added a feature to filter documents in Mongoloader
    - **Feature:** the feature #18251
    - **Dependencies:** No
    - **Twitter handle:** https://twitter.com/im_Kushagra

### CODE REVIEW

**Comment by iamKushagra** [comment]
@baskaryan

File: libs/community/langchain_community/document_loaders/mongodb.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def __init__(
         connection_string: str,
         db_name: str,
         collection_name: str,
+        field_names: Optional[List[str]] = None,
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/mongodb.py:20]: Given that this is a named argument could you put it after the `*` and change the type to Optional[Sequence[str]]

File: libs/community/langchain_community/document_loaders/mongodb.py [hunk 2]
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
...
                 "collection": self.collection_name,
             }
-            result.append(Document(page_content=str(doc), metadata=metadata))
+            if self.field_names is not None:
+                """Filter fields based on field_names"""
+                try:
+                    fields = {name: doc[name] for name in self.field_names}
+                except KeyError as err:
+                    logger.warning(f"{err.args[0]} field not found in Mongo document.")
+                    continue  """Skip this document if a specified field is not found"""
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/mongodb.py:77]: `"""` is used for doc-strings for functions / classes, not for in code documentation

```suggestion
                    continue  # Skip this document if a specified field is not found
```

seme for changes below

File: libs/community/langchain_community/document_loaders/mongodb.py [hunk 3]
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
                 "database": self.db_name,
                 "collection": self.collection_name,
             }
-            result.append(Document(page_content=str(doc), metadata=metadata))
+            if self.field_names is not None:
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/mongodb.py:71]: Is there a way to provide the field names as part of the request to mongo instead? So we don't end up sending those fields over the wire if we don't need them.

File: libs/community/langchain_community/document_loaders/mongodb.py [hunk 4]
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
...
+                """Filter fields based on field_names"""
+                try:
+                    fields = {name: doc[name] for name in self.field_names}
+                except KeyError as err:
+                    logger.warning(f"{err.args[0]} field not found in Mongo document.")
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/mongodb.py:76]: How frequently are fields expected to be missing?

Comment (iamKushagra) [reply libs/community/langchain_community/document_loaders/mongodb.py:76]: There could be instances when in a few documents there might that field doesn't exist. So, the frequency is still low but in any such case, we can skip those.

File: libs/community/langchain_community/document_loaders/mongodb.py (hunk 1 above)
Comment (iamKushagra) [reply libs/community/langchain_community/document_loaders/mongodb.py:20]: Good point!

File: libs/community/langchain_community/document_loaders/mongodb.py (hunk 2 above)
Comment (iamKushagra) [reply libs/community/langchain_community/document_loaders/mongodb.py]: Fixed

File: libs/community/langchain_community/document_loaders/mongodb.py (hunk 1 above)
Comment (iamKushagra) [reply libs/community/langchain_community/document_loaders/mongodb.py]: Fixed

**Comment by eyurtsev** [comment]
This looks good to me -- if you're able to resolve linting error (`make format`) we can merge

**Comment by iamKushagra** [comment]
Thanks @eyurtsev 🚀

**Review by baskaryan — APPROVED** [review]

**Pull request merged**